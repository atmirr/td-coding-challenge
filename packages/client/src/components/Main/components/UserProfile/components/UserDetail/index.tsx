import React from 'react';
import { Link } from 'react-router-dom';
import { makeStyles } from '@material-ui/core/styles';
import { Card, CardContent, Typography, Grid, Box } from '@material-ui/core';
import { ArrowBack } from '@material-ui/icons';
import { UserDetailType } from '../../index';

const useStyles = makeStyles(({ palette, spacing }) => ({
  navigation: {
    paddingBottom: spacing(2),
  },
  backButton: {
    display: 'flex',
    alignItems: 'center',
    textDecoration: 'none',
  },
  arrowBackIcon: {
    color: palette.text.primary,
  },
  backLabel: {
    paddingLeft: spacing(0.5),
    color: palette.text.primary,
  },
  detailsContainer: {
    paddingTop: spacing(2),
  },
}));

function UserDetail({
  name,
  username,
  rating,
  favorites,
  visits,
  size,
  location,
}: UserDetailType) {
  const classes = useStyles();
  return (
    <>
      <div className={classes.navigation}>
        <Link to="/" className={classes.backButton}>
          <ArrowBack className={classes.arrowBackIcon} />
          <Typography component="span" className={classes.backLabel}>
            Back
          </Typography>
        </Link>
      </div>
      <Card>
        <CardContent>
          <Typography component="h2" variant="h5">
            {name}
          </Typography>
          <Typography variant="subtitle1" color="textSecondary">
            {username}
          </Typography>
          <Grid container className={classes.detailsContainer}>
            <Box>
              <Typography variant="subtitle1" gutterBottom>
                Rating:
              </Typography>
              <Typography variant="subtitle1" gutterBottom>
                Favorites:
              </Typography>
              <Typography variant="subtitle1" gutterBottom>
                Visits:
              </Typography>
              <Typography variant="subtitle1" gutterBottom>
                Size:
              </Typography>
              <Typography variant="subtitle1" gutterBottom>
                Location:
              </Typography>
            </Box>
            <Box pl={3}>
              <Typography variant="subtitle1" gutterBottom>
                {rating}
              </Typography>
              <Typography variant="subtitle1" gutterBottom>
                {favorites}
              </Typography>
              <Typography variant="subtitle1" gutterBottom>
                {visits}
              </Typography>
              <Typography variant="subtitle1" gutterBottom>
                {size}
              </Typography>
              <Typography variant="subtitle1" gutterBottom>
                {location}
              </Typography>
            </Box>
          </Grid>
        </CardContent>
      </Card>
    </>
  );
}

export default UserDetail;
