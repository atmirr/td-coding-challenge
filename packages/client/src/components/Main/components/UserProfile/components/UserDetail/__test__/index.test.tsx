import React from 'react';
import '@testing-library/jest-dom';
import '@testing-library/jest-dom/extend-expect';
import { Route, MemoryRouter } from 'react-router-dom';
import { screen, render, fireEvent } from '@testing-library/react';
import UserDetail from '../index';

const props = {
  name: 'Sunny Wucker',
  username: 'madisyn.wuckert51',
  rating: 1.6,
  favorites: 5,
  visits: 65,
  size: 'medium',
  location: '-24.3906 179.7276',
};

describe('<UserDetail />', () => {
  it('should renders the component', () => {
    const { container } = render(
      <MemoryRouter initialEntries={['/user/1']}>
        <UserDetail />
      </MemoryRouter>,
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should display the labels', () => {
    render(
      <MemoryRouter initialEntries={['/user/1']}>
        <UserDetail />
      </MemoryRouter>,
    );
    expect(screen.queryAllByText(/Rating/).length).toBe(1);
    expect(screen.queryAllByText(/Favorites/).length).toBe(1);
    expect(screen.queryAllByText(/Visits/).length).toBe(1);
    expect(screen.queryAllByText(/Size/).length).toBe(1);
    expect(screen.queryAllByText(/Location/).length).toBe(1);
  });

  it("should display the detail's text when we pass the props", () => {
    render(
      <MemoryRouter initialEntries={['/user/1']}>
        <UserDetail {...props} />
      </MemoryRouter>,
    );
    expect(screen.queryAllByText(props.name).length).toBe(1);
    expect(screen.queryAllByText(props.username).length).toBe(1);
    expect(screen.queryAllByText(props.rating).length).toBe(1);
    expect(screen.queryAllByText(props.favorites).length).toBe(1);
    expect(screen.queryAllByText(props.visits).length).toBe(1);
    expect(screen.queryAllByText(props.size).length).toBe(1);
    expect(screen.queryAllByText(props.location).length).toBe(1);
  });

  it('clicking on the back button should take us to the first page', () => {
    let testLocation;
    render(
      <MemoryRouter initialEntries={['/user/1']}>
        <UserDetail />
        <Route
          path="*"
          render={({ location }) => {
            testLocation = location;
            return null;
          }}
        />
      </MemoryRouter>,
    );
    const backLink = screen.getByText(/Back/);
    fireEvent.click(backLink);
    expect(testLocation.pathname).toBe('/');
  });
});
