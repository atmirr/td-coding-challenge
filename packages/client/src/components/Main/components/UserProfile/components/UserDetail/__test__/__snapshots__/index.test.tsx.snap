// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserDetail /> should renders the component 1`] = `
<div
  class="makeStyles-navigation-1"
>
  <a
    class="makeStyles-backButton-2"
    href="/"
  >
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root makeStyles-arrowBackIcon-3"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
      />
    </svg>
    <span
      class="MuiTypography-root makeStyles-backLabel-4 MuiTypography-body1"
    >
      Back
    </span>
  </a>
</div>
`;
